Feature/testimonials

tynandebold opened the pull request:
## Description

Resolves #10.

## Development notes

The testimonials section is a carousel. I've used Framer Motion for the animation and have based it on [this example](https://codesandbox.io/s/framer-motion-image-gallery-pqvx3?file=/src/Example.tsx).

Since this is static content, I haven't added any tests.

I also added a `Makefile` with a `sign-off` command so we don't have to deal with the DCO check anymore.

## QA notes

**Please note** that the implementation doesn't exactly match the layout in the Figma file. We decided to place the control dots underneath the carousel slides on all screen sizes, not just on mobile, as it shows in Figma.

Please refer to [this prototype](https://www.figma.com/proto/3OU7t2Pk3t1pO5eaVi3z0M/Kedro-Prototypes?node-id=421%3A204&scaling=min-zoom&page-id=398%3A70&starting-point-node-id=398%3A182) to see what the design should look like on medium and large screen sizes and to see an approximation of the animation.

Another note: the animation doesn't exactly match the prototype. It follows it closely and makes some subtle tweaks that feel better when swiping to change the slides.

Lastly, I don't have the correct user photos or company logos at the time of writing. I don't believe that's enough to prevent me from opening this PR, so when I get the assets, I'll update the site and this text accordingly.

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Added tests to cover my changes, where applicable

rashidakanchwala reviewed (approved):
thanks @tynandebold.

There is a tiny layout issue I noticed in the mobile mode that happens during transition. I have discuss this with Tynan over slack. This issue is inconsistent and doesn't happen all the time and also we weren't able to replicate it at Tynan's end. Besides that, it work's well hence approving it!

studioswong commented on components/testimonials/testimonials.module.scss:
```diff
@@ -0,0 +1,162 @@
+
+.carousel {
+  display: flex;
+  // overflow: hidden;
```
this can be deleted

studioswong reviewed (changes requested):
great job setting this up, the section works well generally on desktop. I like that you separated the content from the component. 

On testing this on the ipad and phone breakpoints, I notice that layout issues when clicking through the various testimonial cards ( see below) - I think this potentially is the same issue that @rashidakanchwala pointed out above. 

![testimonial ipad air issue](https://user-images.githubusercontent.com/27775658/159671122-91b6ae4c-26e2-4d6a-96ff-df45ddd569cd.gif)
![testimonial iphone animation issue](https://user-images.githubusercontent.com/27775658/159671156-214a1316-bd82-4dc6-be24-cce2ba8bb2e7.gif)

This is likely to be something that is caused by the browser, yet I had tried fresh reloading this on my browser under the mobile breakpoint and it persists. It might be worth to quickly deploy this somewhere so we can directly try this on mobile just to make sure. 
 
I also noticed some problems with the content, where the profile images seems to be really small ( Do we have higher-res images for those?), while one of the link redirects to an invalid link, and another one  (Artem Seleznev's link) redirects to a set of non-english linkedin posts. Would be worth checking with design to double check on obtaining the latest content. 

Update: I've tested this against Safari for the different breakpoints and the animation transition works fine, so this might be simply a chrome browser issue - it would be nice to find out why that is happening on chrome ( and perhaps the best way to find out is to test this directly on a chrome browser on a mobile device.) 

On another note, I notice that the hero image link seemed to be broken on safari ( this does not happen on chrome):
<img width="832" alt="image" src="https://user-images.githubusercontent.com/27775658/159691646-2d6376d7-9ff9-4113-a6d5-8c015943f20c.png">

studioswong commented:
Also one thing I observed that is not relevant to this PR, but would be nice to use this case to address this if possible:

I notice that the 'case-studies' section on mobile still introduces the 3rem border, however that is not the expected behaviour - this is caused by the border that was merged in an earlier PR for the footer component (I've made a note on this change, please see my comment [here](https://github.com/kedro-org/kedro-website/pull/18/files#r833075654))

tynandebold commented:
> 

Good call. I'll remove this.

tynandebold commented:
> great job setting this up, the section works well generally on desktop. I like that you separated the content from the component.
> 
> On testing this on the ipad and phone breakpoints, I notice that layout issues when clicking through the various testimonial cards ( see below) - I think this potentially is the same issue that @rashidakanchwala pointed out above.
> 
> 
> This is likely to be something that is caused by the browser, yet I had tried fresh reloading this on my browser under the mobile breakpoint and it persists. It might be worth to quickly deploy this somewhere so we can directly try this on mobile just to make sure.
> 
> I also noticed some problems with the content, where the profile images seems to be really small ( Do we have higher-res images for those?), while one of the link redirects to an invalid link, and another one (Artem Seleznev's link) redirects to a set of non-english linkedin posts. Would be worth checking with design to double check on obtaining the latest content.
> 
> Update: I've tested this against Safari for the different breakpoints and the animation transition works fine, so this might be simply a chrome browser issue - it would be nice to find out why that is happening on chrome ( and perhaps the best way to find out is to test this directly on a chrome browser on a mobile device.)
> 
> On another note, I notice that the hero image link seemed to be broken on safari ( this does not happen on chrome).

Excellent, thanks for the feedback!

I've addressed the layout shifting on smaller screens. Thankfully that was just one line of CSS. I don't see the issue you're seeing on Safari though. The image loads fine for me. @rashidakanchwala can you please check if you see the Kedro logo in the header in Safari?

Content wise,  I've updated the broken link. For the images in the section, please refer to what I said in the **QA notes** part of the PR description.

studioswong commented on components/testimonials/testimonials.tsx:
```diff
@@ -0,0 +1,160 @@
+                  {testimonials[index].headline}
+                </h3>
+                <p className={style.quote}>
+                  &#34;{testimonials[index].text}&#34;
```
using a `<q>` tag would be more readable here

tynandebold replied:
Nice. Changed.

studioswong reviewed (commented):
Thanks for putting the fix in, I see that it's working well now, well done on finding the fix on that issue 👍 

I have some general comments as listed below that I think would be worth a look. 

In terms of adding test, I still do think that there is value in adding tests even though we are feeding it static content - the component itself is still a dynamic component, and it would be great to add some basic tests to cover and make sure that it is rendering the elements as it should ( such as the dots under the profile, etc). It would be even better if we can test the swipe conditions for the framer motion animation, which I understand how hard it can be to test them.

tynandebold commented on components/testimonials/testimonials.module.scss:
```diff
@@ -0,0 +1,162 @@
+
+.userImage {
+  flex-shrink: 0;
+  height: 394px;
```
Good call.

tynandebold commented:
> Thanks for putting the fix in, I see that it's working well now, well done on finding the fix on that issue 👍
> 
> I have some general comments as listed below that I think would be worth a look.
> 
> In terms of adding test, I still do think that there is value in adding tests even though we are feeding it static content - the component itself is still a dynamic component, and it would be great to add some basic tests to cover and make sure that it is rendering the elements as it should ( such as the dots under the profile, etc). It would be even better if we can test the swipe conditions for the framer motion animation, which I understand how hard it can be to test them.

Roger that. I've remedied your comments and added some tests. Let me know what you think.

studioswong reviewed (approved):
yayy for adding tests - thanks for pushing this through! Great stuff 👍
